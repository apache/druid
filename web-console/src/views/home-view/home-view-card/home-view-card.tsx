/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Card, H5, Icon, IconName } from '@blueprintjs/core';
import classNames from 'classnames';
import React, { ReactNode } from 'react';

import './home-view-card.scss';

export interface HomeViewCardProps {
  className: string;
  onClick?: () => void;
  href?: string;
  icon: IconName;
  title: string;
  loading: boolean;
  error: Error | undefined;
  children?: ReactNode;
}

export const HomeViewCard = React.memo(function HomeViewCard(props: HomeViewCardProps) {
  const { className, onClick, href, icon, title, loading, error, children } = props;

  return (
    <a
      className={classNames('home-view-card', className)}
      onClick={onClick}
      href={href}
      target={href && href[0] === '/' ? '_blank' : undefined}
    >
      <Card interactive elevation={1}>
        <H5>
          <Icon color="#bfccd5" icon={icon} />
          &nbsp;{title}
        </H5>
        {loading ? <p>Loading...</p> : error ? `Error: ${error.message}` : children}
      </Card>
    </a>
  );
});
