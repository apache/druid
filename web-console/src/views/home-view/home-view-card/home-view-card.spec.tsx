/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { IconNames } from '@blueprintjs/icons';
import { render } from '@testing-library/react';
import React from 'react';

import { HomeViewCard } from './home-view-card';

describe('home view card', () => {
  it('matches snapshot', () => {
    const homeViewCard = (
      <HomeViewCard
        className="some-card"
        href={'#somewhere'}
        icon={IconNames.DATABASE}
        title={'Something'}
        loading={false}
        error={undefined}
      >
        Thigns
      </HomeViewCard>
    );

    const { container } = render(homeViewCard);
    expect(container.firstChild).toMatchSnapshot();
  });
});
