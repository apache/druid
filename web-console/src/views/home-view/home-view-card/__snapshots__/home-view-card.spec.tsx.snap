// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home view card matches snapshot 1`] = `
<a
  class="home-view-card some-card"
  href="#somewhere"
>
  <div
    class="bp3-card bp3-interactive bp3-elevation-1"
  >
    <h5
      class="bp3-heading"
    >
      <span
        class="bp3-icon bp3-icon-database"
        icon="database"
      >
        <svg
          data-icon="database"
          fill="#bfccd5"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            database
          </desc>
          <path
            d="M8 4c3.31 0 6-.9 6-2s-2.69-2-6-2C4.68 0 2 .9 2 2s2.68 2 6 2zm-6-.48V8c0 1.1 2.69 2 6 2s6-.9 6-2V3.52C12.78 4.4 10.56 5 8 5s-4.78-.6-6-1.48zm0 6V14c0 1.1 2.69 2 6 2s6-.9 6-2V9.52C12.78 10.4 10.56 11 8 11s-4.78-.6-6-1.48z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      Â 
      Something
    </h5>
    Thigns
  </div>
</a>
`;
